karapace/serialization.py

```diff
@@ -0,0 +1,166 @@
+
+START_BYTE = 0x13
+HEADER_FORMAT = ">bI"
+HEADER_SIZE = 5
+
+
+class InvalidMessageHeader(Exception):
+    pass
+
+
+class InvalidPayload(Exception):
+    pass
+
+
+class InvalidMessageSchema(Exception):
+    pass
+
+
+class SchemaError(Exception):
+    pass
+
+
+class SchemaRetrievalError(SchemaError):
+    pass
+
+
+class SchemaUpdateError(SchemaError):
+    pass
+
+
+def topic_name_strategy(topic_name, record_name):
+    # pylint: disable=W0613
+    return topic_name
+
+
+def record_name_strategy(topic_name, record_name):
+    # pylint: disable=W0613
+    return record_name
+
+
+def topic_record_name_strategy(topic_name, record_name):
+    return topic_name + "-" + record_name
+
+
+NAME_STRATEGIES = {
+    "topic_name": topic_name_strategy,
+    "record_name": record_name_strategy,
+    "topic_record_name": topic_record_name_strategy
+}
+
+
+class SchemaRegistryClient:
+    def __init__(self, schema_registry_url: str = "http://localhost:8081", loop=None):
+        self.client = Client(server_uri=schema_registry_url, client=aiohttp.ClientSession(loop=loop))
+        self.base_url = schema_registry_url
+
+    async def post_new_schema(self, subject, schema):
+        payload = {"schema": json_encode(schema.to_json())}
+        result = await self.client.post("subjects/%s/versions" % quote(subject), json=payload)
+        if result.status_code != 200:
+            raise SchemaRetrievalError(result.json())
+        return result.json()["id"]
+
+    async def get_latest_schema(self, subject):
+        result = await self.client.get("subjects/%s/versions/latest" % quote(subject))
+        if result.status_code != 200:
+            raise SchemaRetrievalError(result.json())
+        json_result = result.json()
+        if "id" not in json_result or "schema" not in json_result:
+            raise SchemaRetrievalError("Invalid result format: %r" % json_result)
+        try:
+            return json_result["id"], avro.io.schema.parse(json_result["schema"])
+        except avro.io.schema.SchemaParseException as e:
+            raise SchemaRetrievalError("Failed to parse schema string from response: %r" % json_result) from e
+
+    async def get_schema_for_id(self, schema_id):
+        result = await self.client.get("schemas/ids/%d" % schema_id)
+        if result.status_code != 200:
+            raise SchemaRetrievalError(result.json()["message"])
+        json_result = result.json()
+        if "schema" not in json_result:
+            raise SchemaRetrievalError("Invalid result format: %r" % json_result)
+        try:
+            return avro.io.schema.parse(json_result["schema"])
+        except avro.io.schema.SchemaParseException as e:
+            raise SchemaRetrievalError("Failed to parse schema string from response: %r" % json_result) from e
+
+
+class SchemaRegistrySerializerDeserializer:
+    def __init__(self, **config):
+        if not config.get("config_path"):
+            raise Exception("config_path is empty or missing")
```

Comment:
We might want to raise something specific here ValueError or RuntimeError perhaps? Depending on use case?

Reply:
fixed